src/libstd/collections/hash/map.rs

```diff
@@ -1271,6 +1299,12 @@ pub struct Drain<'a, K: 'a, V: 'a> {
     inner: iter::Map<table::Drain<'a, K, V>, fn((SafeHash, K, V)) -> (K, V)>
 }
 
+/// Mutable HashMap values iterator.
+#[unstable(feature = "map_values_mut", reason = "recently added", issue = "32551")]
+pub struct ValuesMut<'a, K: 'a, V: 'a> {
+    inner: Map<IterMut<'a, K, V>, fn((&'a K, &'a mut V)) -> &'a mut V>
```

Comment:
Is there any reason that you chose to use `Map` here and not in `BTreeMap`'s `ValuesMut`? I don't think that there are variance issues (since `&mut` forces `V` to be invariant anyway), but the inconsistency seems a little weird.